### PR TITLE
Expand GCM runs to full footprint

### PR DESCRIPTION
**Pull request by hcorson-dosch-usgs**
These code changes allow for expansion of the GCM runs to the full model footprint.

Specific changes:
* Remove filtering to GLM-modelable lakes, since now completed [in `lake-temperature-model-prep`](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/_targets.R#L60)
* Add a step to also restrict GCM evaluation to only sites within CASC states

To actually run for the full footprint, this requires manually transferring [these files](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/7_drivers_munge/out/7_GCM_driver_files.ind#L1-L7) updated by the expansion of the [ `lake-temperature-model-prep` `targets` sub-pipeline](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/_targets.R) to the `'1_prep/in'` folder of this pipeline:
* `'1_prep/in/lake_cell_tile_xwalk.csv'`
* `'1_prep/in/GCM_{gcm name}.nc'` (6 netCDF files)

### CODE REVIEW

File: 5_evaluate.R [hunk 1]
```diff
@@ -2,13 +2,20 @@ source('5_evaluate/src/eval_utility_fxns.R')
 source('4_visualize/src/plot_data_utility_fxns.R')
 
 p5 <- list(
+  # For now, evaluate only predictions for lakes within CASC states
+  tar_target(p5_CASC_states,
+             c('ND','SD','IA','MI','IN','IL','WI','MN','MO','AR','OH')),
```
Comment (lindsayplatt) [inline 5_evaluate.R:7]: Is this also being applied to what we are bundling up somewhere?

File: 1_prep.R [hunk 2]
```diff
@@ -25,14 +25,12 @@ p1 <- list(
                filter(site_id %in% p1_nml_site_ids) %>%
                arrange(site_id)),
 
-  # lake - GCM cell - GCM tile crosswalk, filtered to only those lakes
-  # that are in nml list (able to be modeled by GLM)
+  # lake - GCM cell - GCM tile crosswalk
```
Comment (lindsayplatt) [inline 1_prep.R]:
```suggestion
  # lake - GCM cell - GCM tile crosswalk, assumed to only include lakes
  # that are in the nml list (able to be modeled by GLM)
```

**Review by lindsayplatt — APPROVED** [review]

File: 5_evaluate.R (hunk 1 above)
Comment (hcorson-dosch-usgs) [reply 5_evaluate.R:7]: No, not at the moment. Do we want to restrict that as well?

Comment (hcorson-dosch-usgs) [reply 5_evaluate.R:7]: At the moment, NLDAS output is zipped by state, so is easy for the user to partition if they only want results for certain states (and we're manually uploading only zips for CASC states). The GCM output for now is zipped by tile, so is less simple to partition to CASC states.

Comment (lindsayplatt) [reply 5_evaluate.R:7]: OK, just thinking about when we go to upload how there is a bit of a manual step right now

**Pull request merged**